### PR TITLE
Deprecate isfloat

### PR DESCRIPTION
**Pull request by stonea**
This PR addresses https://github.com/chapel-lang/chapel/issues/19362 by deprecating the `isFloat`, `isFloatValue`, and `isFloatType` functions.

These are the warning messages users will get now if they attempt to use these functions:

```
warning: isFloat is deprecated use `isReal(e) || isImag(e)` instead
warning: isFloatType is deprecated use `isRealType(t) || isImagType(t)` instead
warning: isFloatValue is deprecated use `isRealValue(e) || isImagValue(e)` instead
```

Why deprecate these?

* It's confusing that it returns true for real and imag but not complex.
* It doesn't appear to be heavily used anyway
* Maybe someday we'll add a builtin generic float or floatingPoint typeclass (similar to what we have with integral or numeric) in which case that might influence what we'd want some future version of isFloat to mean.

Implementing this change involved updating a handful of places that `isFloat` is used in our standard module code as well as updating some tests.

### CODE REVIEW

**Review by jeremiah-corrado — APPROVED** [review]
Overall, I think this looks good! I just had a few minor questions/suggestions.

File: modules/standard/IO.chpl [hunk 1]
```diff
@@ -3379,7 +3379,7 @@ private proc _read_binary_internal(_channel_internal:qio_channel_ptr_t, param by
     } else {
       compilerError("Unknown int type in _read_binary_internal ", t:string);
     }
-  } else if isFloatType(t) {
+  } else if isRealType(t) || isImagType(t) {
```
[the remark this answers is not in the document]
Comment (stonea) [reply modules/standard/IO.chpl:3382]: I think the idea behind that compiler error is to serve as a kind of assertion that would get triggered if we added a new type but failed to add a new branch to handle it.

Say if we added `real(16)` or `real(128)` (or same thing with `imag`) someday.

File: test/deprecated/typeQueryFuncs.chpl [hunk 2]
```diff
@@ -55,3 +55,10 @@ writeln("op <= returns ", BaseClass <= DerivedClass);
 writeln("op <  returns ", BaseClass <  DerivedClass);
 writeln("op >= returns ", BaseClass >= DerivedClass);
 writeln("op >  returns ", BaseClass >  DerivedClass);
+
+// isFloat* functions are being deprecatd
+// (in favor of doing isReal(x) || isImag(x) instead)
+
+writeln("isFloat(1.2) returns ", isFloat(1.2));
```
[the remark this answers is not in the document]
Comment (stonea) [reply test/deprecated/typeQueryFuncs.chpl:62]: Done.

File: test/users/vass/isX/src/isX.byBlankArgs-compWarns.decls-chpl [hunk 3]
```diff
@@ -21,7 +21,7 @@ const hetInstance = (101, 102.5, "hetInstance");
 class ClassType {
   var zzz: int;
 }
-const cInstance = new ClassType(44444444);
+const cInstance = (new owned ClassType(44444444)).borrow();
```
[the remark this answers is not in the document]
Comment (stonea) [reply test/users/vass/isX/src/isX.byBlankArgs-compWarns.decls-chpl:24]: More putting something that should have been added from a previous (unrelated) PR but wasn't.

This file is a template that's used to produce another `.chpl` file. Both the template and the produced file are checked into version control.

The previous PR updated the produced file but not the template (see here https://github.com/chapel-lang/chapel/commit/b665e0c7686ebbd689c411a6e7454832f5a93d4e#diff-9a6a1ea837cbdd16931115143d7209a37a08d64b4cf986336a1949186334083d).

So I updated it in this PR. I should have pointed this out in my PR description but forgot to, sorry about that.

File: test/users/vass/isX/src/isX.module-writeln.decls-chpl [hunk 4]
```diff
@@ -23,7 +23,7 @@ const hetInstance = (101, 102.5, "hetInstance");
 class ClassType {
   var zzz: int;
 }
-const cInstance = new ClassType(44444444);
+const cInstance = (new owned ClassType(44444444)).borrow();
```
[the remark this answers is not in the document]
Comment (stonea) [reply test/users/vass/isX/src/isX.module-writeln.decls-chpl:26]: Another example of updating the template file to fix something that was missing from a prior PR.

File: test/users/vass/isX/isX.byBlankArgs-compWarns.na-none.good [hunk 5]
```diff
@@ -161,67 +161,55 @@ isNumericType
...
 isPrimitiveValue
 isPrimitiveType
-  isX.byBlankArgs-compWarns.chpl:203: called as test(arg: uint(64))
+  isX.byBlankArgs-compWarns.chpl:200: called as test(arg: uint(64))
 
 real(32)
 isReal
 isRealValue
 isRealType
-isFloat
```
[the remark this answers is not in the document]
Comment (stonea) [reply test/users/vass/isX/isX.byBlankArgs-compWarns.na-none.good:170]: Looking at our test system doc, it looks like `.na-none.good` files get run when `CHPL_NETWORK_ATOMICS=none`.  https://github.com/chapel-lang/chapel/blob/main/doc/rst/developer/bestPractices/TestSystem.rst

Running `start_tests .` everything passes but maybe we have some test configuration that would use the other file so I think you're right I should update `isX.byBlankArgs-compWarns.chpl`. 

Thanks for noticing and pointing this out

File: modules/standard/IO.chpl (hunk 1 above)
[the remark this answers is not in the document]
Comment (jeremiah-corrado) [reply modules/standard/IO.chpl:3382]: Ok that makes sense. Just wanted to make sure we weren't overlooking something.

**Pull request merged**